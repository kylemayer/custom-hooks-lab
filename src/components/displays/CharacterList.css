.list {
    list-style-type: none;
}

.caption {
    text-align: center;
}