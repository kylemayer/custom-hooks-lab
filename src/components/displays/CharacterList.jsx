import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import { Link } from 'react-router-dom';
import styles from './CharacterList.css';

const CharacterList = ({ characters }) => (
  <ul aria-label="characters">
    {characters.map((character) => (
      <Link key={character.id} to={`/character/${character.id}`}>
        <li className={styles.list} key={character.id}>
          <Character {...character} />
        </li>
      </Link>
    ))}
  </ul>
);

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
    }).isRequired
  ),
};

export default CharacterList;
