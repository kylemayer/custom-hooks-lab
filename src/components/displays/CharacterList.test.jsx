import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('Character list component', () => {
  afterEach(() => cleanup());
  it('displays a list of hey arnold characters', async () => {
    const { asFragment } = render(
      <CharacterList characters={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
