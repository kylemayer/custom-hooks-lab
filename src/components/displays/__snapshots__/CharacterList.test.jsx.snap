// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character list component displays a list of hey arnold characters 1`] = `
<DocumentFragment>
  <ul
    aria-label="characters"
  />
</DocumentFragment>
`;
